pkg/daemon/update.go

```diff
@@ -1147,42 +1147,57 @@ func (dn *Daemon) switchKernel(oldConfig, newConfig *mcfgv1.MachineConfig) error
 
 	defaultKernel := []string{"kernel", "kernel-core", "kernel-modules", "kernel-modules-extra"}
 	realtimeKernel := []string{"kernel-rt-core", "kernel-rt-modules", "kernel-rt-modules-extra", "kernel-rt-kvm"}
-	var args []string
 
 	dn.logSystem("Initiating switch from kernel %s to %s", canonicalizeKernelType(oldConfig.Spec.KernelType), canonicalizeKernelType(newConfig.Spec.KernelType))
 
-	if canonicalizeKernelType(oldConfig.Spec.KernelType) == ctrlcommon.KernelTypeRealtime && canonicalizeKernelType(newConfig.Spec.KernelType) == ctrlcommon.KernelTypeDefault {
-		args = []string{"override", "reset"}
+	// resetRealTimeOverride removes the realtime kernel override.
+	resetRealTimeOverride := func(msg string) error {
```

Comment:
override reset is used only for base overrides (packages which is part of base OS). In our case we override regular kernel packages which is part of base OS. kernnel-rt packages are layered packages and are treated similar to any other layered packages. Layered packages should automatically updated from extensions repo when `rpm-ostree update` gets called.

Reply:
> Sigh, this regression occurred due to [adb9e70#diff-349c0748c3f52201852d3027c29daf618b45300b949482728df6666a3f9ba245R963](https://github.com/openshift/machine-config-operator/commit/adb9e707b0d0170d741cc40dec2eaa73aa201415#diff-349c0748c3f52201852d3027c29daf618b45300b949482728df6666a3f9ba245R963) . I blame copy paste

Ah, you found the real culprit. `rpm-ostree`'s man page doesn't document `rpm-ostree update` (I went looking).

Reply:
yup, it is tricky